girder-dandi-archive/girder_dandi_archive/rest.py

```diff
@@ -0,0 +1,64 @@
+from girder.api import access
+from girder.api.rest import Resource
+from girder.api.describe import describeRoute, Description
+from girder.models.setting import Setting
+from girder.models.folder import Folder
+from girder.exceptions import RestException
+
+from .util import (
+    DANDISET_ID_COUNTER,
+    staging_collection,
+    pad_dandiset_id,
+    validate_dandiset_id,
+)
+
+
+class DandiResource(Resource):
+    def __init__(self):
+        super(DandiResource, self).__init__()
+
+        self.resourceName = "dandi"
+        self.route("GET", (), self.get_dandiset)
+        self.route("POST", (), self.create_dandiset)
+
+    @access.user
+    @describeRoute(
+        Description("Create Dandiset")
+        .param("name", "Name of the Dandiset.")
+        .param("description", "Description of the Dandiset.")
+    )
+    def create_dandiset(self, params):
+        name, description = params["name"], params["description"]
```

Comment:
`@describeRoute` just updates Swagger docs. It does no validation of `params`.

In the case that a caller fails to pass `name` or `description` in the form body, the server will 500 with a `KeyError`.

Reply:
Good catch